### ISSUE TITLE
Predicted Q values and average loss become exponentially larger

### ISSUE THREAD

**Issue by alan-d-smith** [AlanDoesCS/RL-Pathfinding#3]
### Describe the bug
When attempting to train a DDQN on a grid environment, the average loss/episodes appears to exponentially increase, and the magnitude of the predicted Q values also rapidly increases, even with (relatively) simple network architectures

### To Reproduce
- Run this section of code:
```java
public static void main(String[] args) {
        Environment.setStateType(Environment.StateType.PositionVectorOnly);
        Environment.setDimensions(10, 10);
        Environment.setActionSpace(4);

        DDQNAgentTrainer trainer;
        try {
            trainer = new DDQNAgentTrainer(Set.of(EmptyGridEnvironment.class));
        } catch (InvalidTypeException e) {
            e.printStackTrace();
            return;
        }

        List<Layer> layers = new ArrayList<>();
        LeakyReLU leakyRelu = new LeakyReLU(0.1f);
        float lambda = 0.0001f;

        layers.add(new MLPLayer(Environment.getStateSpace(), 64, leakyRelu, 0, lambda));
        layers.add(new MLPLayer(64, 64, leakyRelu, 0, lambda));
        layers.add(new MLPLayer(64, Environment.getActionSpace(), new Linear(), 0, lambda));

        DDQNAgent ddqnAgent = new DDQNAgent(
                Environment.getActionSpace(),  // action space
                layers,                        // layers
                1,                             // initial epsilon
                0.9999,                        // epsilon decay
                0.01,                          // epsilon min
                0.999,                         // gamma
                0.0001,                        // learning rate
                0.99995,                       // learning rate decay
                0.00005f,                      // learning rate minimum
                0.005                          // tau
        );

        ddqnAgent.dumpDQNInfo();
        trainer.trainAgent(ddqnAgent, 600000, 500, 1, "plot", "ease", "axis_ticks", "show_path");
    }
```

![Average Loss exponentially increases over time](https://github.com/user-attachments/assets/e35fddeb-2ac7-4a66-8143-e4b63d639dc3)
![Q Values become very large](https://github.com/user-attachments/assets/65a80d13-2e0d-4c39-981c-01e25024d092)

**Comment by alan-d-smith** [comment]
Resolved in:
https://github.com/AlanDoesCS/Easy-Java-RL-Library/commit/f313518d446f308d97a44e350270a4100b1d2370
